searx/shared/__init__.py

```diff
@@ -4,33 +4,47 @@
 
 logger = logging.getLogger('searx.shared')
 
+__all__ = ["storage", "schedule"]
+
+
 try:
-    import uwsgi
-except:
-    # no uwsgi
-    from .shared_simple import SimpleSharedDict as SharedDict, schedule
+    # First: try to use Redis
+    from .redisdb import client
+
+    client().ping()
+    from .shared_redis import RedisCacheSharedDict as SharedDict, schedule
 
-    logger.info('Use shared_simple implementation')
-else:
+    logger.info('Use shared_redis implementation')
+except Exception as e:
+    # Second: try to use uwsgi
     try:
-        uwsgi.cache_update('dummy', b'dummy')
-        if uwsgi.cache_get('dummy') != b'dummy':
-            raise Exception()
+        import uwsgi
     except:
-        # uwsgi.ini configuration problem: disable all scheduling
-        logger.error(
-            'uwsgi.ini configuration error, add this line to your uwsgi.ini\n'
-            'cache2 = name=searxcache,items=2000,blocks=2000,blocksize=4096,bitmap=1'
-        )
-        from .shared_simple import SimpleSharedDict as SharedDict
-
-        def schedule(delay, func, *args):
-            return False
+        # Third : fall back to shared_simple
+        from .shared_simple import SimpleSharedDict as SharedDict, schedule
 
+        logger.info('Use shared_simple implementation')
     else:
-        # uwsgi
-        from .shared_uwsgi import UwsgiCacheSharedDict as SharedDict, schedule
-
-        logger.info('Use shared_uwsgi implementation')
+        # Make sure uwsgi is okay
+        try:
+            uwsgi.cache_update('dummy', b'dummy')
+            if uwsgi.cache_get('dummy') != b'dummy':
+                raise Exception()  # pylint: disable=raise-missing-from
+        except:
+            # there is exception on a get/set test: disable all scheduling
+            logger.exception(
+                'uwsgi.ini configuration error, add this line to your uwsgi.ini\n'
+                'cache2 = name=searxcache,items=2000,blocks=2000,blocksize=4096,bitmap=1'
```

Comment:
name of the cache --> `searxngcache` / https://github.com/searxng/searxng-docker/issues/31#issuecomment-1177418360

Reply:
fixed